Update asset fields and make metadata doc less about assets (#25)
radiantearth/stac-best-practices: best-practices-metadata-and-extension.md

    Note that the Storage Extension is still evolving and has a limited set of cloud stores defined.
    Please open a PR if your cloud store is not available yet.

> 

Reviewer: I don't think I agree with removing "Items" and "Assets" as sections. Several of the properties below, I've never seen used in the Item and the following section would imply for me having them at the item level (data_type, offset, scale, classification:bit_fields, etc.)

## General fields

- **Common metadata**
  - `datetime` Provide individual timestamp on an Item, in case the Item has a `start_datetime` and `end_datetime`,

>   but an Asset is for one specific time.

Reviewer: I'm not even sure I understand this yet...?!

  
    `proj:code` can be set to `null` if one of the other two fields is provided.
    `proj:wkt2` and `proj:projjson` should never be set to `null`.

    Typically specified at the Item level. If the projection is different
    for all assets it should not be provided as an Item property. If most assets are one projection, and there is 
    a single reprojected version (such as a Web Mercator preview image), it is sensible to specify the main projection in the 
    Item and the alternate projection for the affected asset(s).

  - Two of the following: `proj:shape`, `proj:transform`, `proj:bbox`.
  
    `proj:bbox` should be omitted if the values match `bbox` property exactly (i.e. if `proj:code` "is a geographic coordinate reference system, using the World Geodetic System 1984 (WGS 84) datum, with longitude and latitude units of decimal degrees" (see [RFC 7946, ch. 4](https://datatracker.ietf.org/doc/html/rfc7946#section-4))).

>     Typically specified at the Item level. If assets have different spatial resolutions and slightly different exact bounding boxes,

Reviewer: I actually see it more often specified at the asset level.

```suggestion
    If assets have different spatial resolutions and slightly different bounding boxes,
```

>     specify these per asset to indicate the size of the asset in pixels and its exact GeoTransform in the native projection.

Reviewer: ```suggestion
    specify these per asset to indicate the size of the asset in pixels and its exact affine transform in the native projection.
```

GeoTransform is incorrect!

  
- **Raster Extension** (v2.x)

  - `raster:spatial_resolution` (if multiple resolutions are provided in the Asset, otherwise in the Item)
  - `raster:scale`

>   - `raster:offset`

Reviewer: These are usually used on the asset level, not the Item property level. Thus I find the lack of distinction between Asset and Item a bit confusing.

  - `eo:common_name` (if applicable)
  - `eo:center_wavelength` and `eo:full_width_half_max`

## SAR Measurements

- **[SAR Extension](https://github.com/stac-extensions/sar)**  (v1.2 or a later non-breaking version)

>   - `sar:polarizations` (in the order they should appear in concatenated data structure)

Reviewer: Not sure, but it feels like a word is missing.

```suggestion
  - `sar:polarizations` (in the order they should appear in the concatenated data structure)
```

- **[Product Extension](https://github.com/stac-extensions/product)**
  - `product:type`
   

>     If mixing multiple product types within a single Item, this can be used to specify per asset.

Reviewer: Although it's allowed in assets, a product for me is a item and can't be an individual asset. If the product:type between assets differs, I think it should be a separate item.
